remoteoss/json-schema-form
# fix(modify): Support json schemas without allOf

## What this PR does

Fix an error that occurs if you pick fields using `modify()` on a schema that doesn't define `allOf`.

[Internal ticket](https://linear.app/remote/issue/INF-639)

### Why it happened / how it's fixed

Unguarded access to `allOf.forEach(...)` on the modified schema, which we now check.

### How to reproduce and test

The following example no longer errors on this branch. The added test case demonstrates this.

```js
const schema = {
 // This schema does not use `allOf`
  properties: {
    age: {
      title: 'Age',
      type: 'integer',
    },
    quantity: {
      title: 'Quantity',
      type: 'integer',
    },
  },
  'x-jsf-order': ['age', 'quantity'],
};

// TypeError reading properties of allOf
modify(schema, {
  pick: 'age'
})
```
